hnn_core/feed.py

```diff
@@ -2,65 +2,82 @@
 
 # Authors: Mainak Jas <mainak.jas@telecom-paristech.fr>
 #          Sam Neymotin <samnemo@gmail.com>
+#          Christopher Bailey <bailey.cj@gmail.com>
 
 import numpy as np
 from neuron import h
 
 
 class ExtFeed(object):
-    """"The ExtFeed class.
+    """"The ExtFeed class of external spike input times.
+
+    An external input "feed" to the network, i.e., one that is independent of
+    the spiking output of cells in the network.
 
     Parameters
     ----------
-    ty : str
-        The feed type. Can be:
-        'extpois', 'evprox', 'evdist', 'extgauss', 'extinput'
-    celltype : str | None
-        The cell type.
-    p_ext : dict | list (XXX: check)
-        Parameters of external input.
-        p_ext has a different structure for the extinput
-        usually, p_ext is a dict of cell types
+    feed_type : str
+        The feed type, which is one of
+        'extpois' : Poisson-distributed input to proximal dendrites
+        'extgauss' : Gaussian-distributed input to proximal dendrites
+        'evprox' : Proximal input at specified time (or Gaussian spread)
+        'evdist' : Distal input at specified time (or Gaussian spread)
+        'common' : Common pulses to proximal and distal dendrites
```

Comment:
Nice thanks! How did you figure out that it's "common"?

I think we should probably get rid of this class and separate these methods into functions: `create_evoked`, `create_uniform` with function signatures that clearly specify what parameters it expects: e.g., t_start, t_std, freq etc.

Reply:
But this could be another PR

Reply:
I was under the impression that what we previously called 'extinput' was a common input to all neurons in at the soma (i.e., a modification of the proximal input) rather than to both the proximal and distal input levels. Is this wrong?

Reply:
It took me some time to wrap my head around this! Create the `net` for the `alpha` example in `master` & look at `net.p_ext` (renamed in this PR): it's a `len==2` list, where each dict has the key `'loc'` being `'distal'` or `'proximal'`.

Figure 3 of the [alpha/beta tutorial](https://jonescompneurolab.github.io/hnn-tutorials/alpha_and_beta/alpha_and_beta) has the schematic. I didn't actually follow the code (yet) to confirm this, but there's a lot of 'evidence' to these being dendritic inputs for pyramidals & somatic for baskets, though for some reason [L5 baskets are not included in the parameter list](https://github.com/cjayb/hnn-core/blob/822d23945ff2f0907b28896d1d329be6bd271742/hnn_core/params.py#L328). Separate investigation/PR...

Reply:
Let me correct my previous statement: an 'extinput' (or what you're labelling 'common') can either be at the proximal location, distal location, or both. The distinguishing factor when comparing it to a proximal evoked input, for example, is that the 'extinput' at the proximal location treats the feed as a single input directly to all L5 pyramidal cells. Evoked inputs, on the other hand, create a separate feed for each 'proximal' cell (i.e., one for each of the 10x10 L5 pyramidal cells, the 10x10 L2/3 pyramidal cells, the 35 L5 basket cells and 35 L2/3 basket cells) dependent on the local network connectivity. @samnemo is this correct?

@cjayb Maybe this is what you were already thinking, I just want to make sure we find a clear way to represent this in the code so that it is less confusing in the future.

Reply:
@rythorpe that’s exactly my understanding too, yes. Whether or not ‘common’ is a good term is debatable. I think the way params are handled (list of dict in one place, dict of dict in another) is a little muddy still. Perhaps @jasmainak ‘s suggestion of refactoring into functions rather than an object (which hardly uses statefulness as it is anyway) will help.

Reply:
I see, so "common" is somehow opposed to "unique"? Could you clarify above in the docstring that it can be either proximal or distal?

also are you able to tell [based on the code](https://github.com/jonescompneurolab/hnn-core/blob/master/hnn_core/network.py#L336-L356) how one is unique to each cell and the other is not?

The list of dict and dict of dict is indeed an issue. Let's open an issue for that so we can understand it better? I would begin by documenting what exactly these data structures do.

Reply:
> also are you able to tell [based on the code](https://github.com/jonescompneurolab/hnn-core/blob/master/hnn_core/network.py#L336-L356) how one is unique to each cell and the other is not?
> 
I find the code to be fairly opaque in this regard.